frank/fit.py

```diff
@@ -0,0 +1,275 @@
+# Frankenstein: 1D disc brightness profile reconstruction from Fourier data
+# using non-parametric Gaussian Processes
+#
+# Copyright (C) 2019-2020  R. Booth, J. Jennings, M. Tazzari
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>
+#
+"""This module runs Frankenstein to fit a source's 1D radial brightness profile.
+   A default parameter file is used that specifies all options to run the fit
+   and output results. Alternatively a custom parameter file can be provided.
+"""
+
+import os
+import sys
+import time
+import json
+import numpy as np
+
+import logging
+
+
+def helper():
+    with open('parameter_descriptions.json') as f:
```

Comment:
This "open" needs the full path to the library. Unless they get installed properly somewhere we are going to have issues.

For the purpose of just having a documentation file for the parameters it might be best to just provide the dictionary directly rather than using a json.

Reply:
That's why I commented inline on this under 'TODO'. 

The parameter descriptions should be a `.json` of the same structure as `default_parameters.py`. If `default_parameters.py` is changed at some point, hardcoded text in `fit.py` is not the place to update the parameter descriptions. There must be a way to import a file from a package; I just haven't found the right solution yet.

Reply:
After our discussion we agree on keeping `parameter_descriptions` as a `.json`, just need to correctly link to it. This is already a `TODO` in `fit.py` Resolving here.